js/packages/cli/src/helpers/constants.ts

```diff
@@ -21,7 +21,8 @@ export const CONFIG_ARRAY_START =
   8 + //max supply
   1 + //is mutable
   1 + // retain authority
-  4; // max number of lines;
+  4 + // max number of lines
+  1; // is templated
```

Comment:
What you have not considered is this will break every candy machine in existence that isn't your own new kind. You need to think of a way to gracefully handle them.

Reply:
Only way I can see this working is with a ConfigV2.

Reply:
Many thanks for your feeback, and damn, you are right, sorry for that!

I'm a bit new to both solana and rust which may explain why I'm not sure what would be the best way to handle backward compatibility in this case. Is the `ConfigV2` you pointed a `ctx.accounts.configv2` dedicated account, that we could check to know if we're dealing with a configuration including the `is_templated` field?

Another way of implementing that could be to set the `max_number_of_lines` to `u32::MAX` when declaring a templated configuration, that value triggering the required mechanisms. It feels a bit hacky but could be less intrusive than a dedicated account.

Anyway, I would be more than happy to implement the preferred solution if this template mechanism is welcome in the basecode of metaplex.

Reply:
Well Configv2 would literally be a new type so it would have a different account type. This would mean, as far as anchor is concerned, new endpoints (you would duplicate them and instead of having Config you would use ConfigV2, and then update the CLI). This would mean that people using the older cli and endpoints would ocntinue experiencing normal operations and those using the newer CLI would exprience no issues either..

Reply:
Got it, I actually implemented it using the latter way (a custom `max_number_of_lines` value triggering the required mechanisms), it is working but it was still needing cleanup and like I was expecting, it feels hacky.

I see here a good opportunity to introduce versioning of the candy machine configuration structure, I should have some time to implement it this way this week-end. Many thanks for your insights!

Reply:
I have been able to implement backward compatbility using an `initialize_config_v2` method, and because duplicating all anchor endpoints felt a bit bloaty, I used the part of the configuration that is manually serialized to add the required information at the very end of the configuration,  this information is checked in a backward compatible way in `is_config_templated()` [here](https://github.com/soualid/metaplex/blob/a9a5e75d6ecc02636db1bb1bdcd9fca69a2de326/rust/nft-candy-machine/src/lib.rs#L567), which check this value, or return `false` if the size of the configuration indicate that we are dealing with a configuration created before the implementation of this feature.

Reply:
Ok i will take a look at this